x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/ModelLoadingService.java

```diff
@@ -507,6 +521,10 @@ private void cacheEvictionListener(RemovalNotification<String, ModelAndConsumer>
                 modelId,
                 modelIdToModelAliases.getOrDefault(modelId, new HashSet<>())
             ));
+            // If it's not referenced in a pipeline, stop tracking it on this node
```

Comment:
Does this mean that usage of inference models in pipeline aggregations but _not_ in ingest pipelines isn't tracked?

If so then we should do something about this, although not necessarily in this PR - it can be deferred to a followup.  Maybe we need to track a "momentary" feature when an inference model is used in a pipeline aggregation, or maybe this already happens simply by us checking the license.  I just want to make sure this case hasn't been missed.

Reply:
@droberts195 I think its fine to remove from tracking here as it is no longer cached. 

Let me clarify what start tracking + stop tracking does for persistent features.

- Start tracking for a persistent feature means when we ask for the "last used" its always "now"
- Stop tracking for a persistent feature means that the "last used" is set to the moment in time its stopped tracking

If we were to use "momentary" feature, it only sets the last used to the exact moment the function "track" function is called. So, the model could be in cache for longer than we say its in use.



So, in the current code, while the model is cached, the "last used" is always "now". Once we evict it, the last used becomes the moment in time `stopTracking` was called. 

We have two cases:

 - In search only, we only say the model is tracked while cached. It flipping in and out of cache is OK, as the caller only cares about the "last used" ticker (which gets overridden on the next time its cached)
 - In pipelines, we say its in use for as long as it is in the pipeline. 

I think this is acceptable. Switching between tracking momentarily vs persistent is complicated...What if a model is "momentarily" used in a search and THEN added to a pipeline?!?!